script/main.js

```diff
@@ -43,65 +43,76 @@ async function showNotebook(){
-        : notebook[i].description;
+    notebook.forEach(note => {
+        const limitedDescription = note.description.length > 100
+        ? note.description.slice(0,30) + "..."
+        : note.description;
 
         notesContainer.innerHTML += 
         `
         <div class="notebook-single-container">
-            <div class="title-div" data-id="${notebook[i].id}">
-                <h5 class="title-container">${notebook[i].title} </h5>
+            <div class="title-div" data-id="${note.id}">
+                <h5 class="title-container">${note.title} </h5>
                 
             </div>
             <div class="notebook-entry">
                 <p class="entry">${limitedDescription}</p>
             </div>
             <div class="options">
                 <span>
-                    <i onClick="editPost(this)" class="fas fa-edit"></i>
+                    <i onClick="editNote(this)" class="fas fa-edit"></i>
                     <i onClick="deleteNote(this)" class="fas fa-trash-alt"></i>
                 </span>
             </div>
         </div>
       `;
       
-    }
+    })
 
 }
 async function deleteNote(){
-    const deleteButtons = document.querySelectorAll('.fa-trash-alt');
-
-    deleteButtons.forEach(button => {
-        button.addEventListener('click', async function (event) {
-            event.stopPropagation();
-            const noteContainer = this.closest('.notebook-single-container');
-            const noteId = noteContainer.firstElementChild.getAttribute('data-id');
-
-            const response = await fetch(`${APIUrl}/delete`,{
-                method: 'DELETE',
-                headers: {
-                    'Content-type':'application/json'
-                },
-                body: JSON.stringify({id:noteId})
-            })
-            if (!response.ok){
-                throw new Error('There is a problem.');
-            }
-            const result = await response.json();
-            location.reload();
-        });
-    });
-    
+    const notesContainer = document.getElementById('notebook-entries');
+
+        notesContainer.addEventListener('click', async function (event) {
+            if(event.target.classList.contains('fa-trash-alt')){
+                event.stopPropagation();
+                const noteContainer = event.target.closest('.notebook-single-container');
+                const noteId = noteContainer.firstElementChild.getAttribute('data-id');
+
+                try{
+                    const response = await fetch(`${APIUrl}/delete`,{
+                    method: 'DELETE',
+                    headers: {
+                        'Content-type':'application/json'
+                    },
+                    body: JSON.stringify({id:noteId})
+                    
+                })
+                    const result = await response.json();
+                    noteContainer.remove()
+                }
+                catch(error){
+                    console.error(error)
```

Comment:
what will do user get when this doesnt work ? the user needs to get an update if it didnt work or when it worked.